### PR TITLE
WIP: EventIngester CRD

### PR DESCRIPTION
**Pull request by suprjinx**
main

### CODE REVIEW

File: controllers/install/eventingester_controller.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+	mutateFn := func() error { return nil }
+
+	if components.ServiceAccount != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, components.ServiceAccount, mutateFn); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	if components.ClusterRole != nil {
```
Comment (kannon92) [inline controllers/install/eventingester_controller.go:85]: Looking at event ingestor chart in Armada, I think you only need a deployment, secret, and a service account.

Comment (kannon92) [reply controllers/install/eventingester_controller.go:86]: The event-ingestor chart has a deployment, secret and a service account.  No service.  

TBH I don't really know what a service account is but that is required even if you don't have a service.

File: apis/install/v1alpha1/common.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ import (
 )
 
 type Image struct {
```
Comment (kannon92) [inline apis/install/v1alpha1/common.go:23]: I don’t think you need this.

Comment (kannon92) [reply apis/install/v1alpha1/common.go:23]: I think I am wrong.  I notice that executor and binoculars don't use Image but we probably do want that.  Don't change this after all.

File: controllers/builders/generate_config_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+			input:    runtime.RawExtension{Raw: []byte(`{ "test": { "foo": "bar" }}`)},
+			expected: "test:\n  foo: bar\n",
+		},
+		{
+			name:     "it converts complex runtime.RawExtension json to yaml",
+			input:    runtime.RawExtension{Raw: []byte(`{ "test": {"foo": "bar"}, "test1": {"foo1": { "foo2": "bar2" }}}`)},
+			expected: "test:\n  foo: bar\ntest1:\n  foo1:\n    foo2: bar2\n",
+		},
+		{
+			name:     "it ghacks if runtime.RawExtension raw is malformed json",
```
Comment (kannon92) [inline controllers/builders/generate_config_test.go:30]: typo?

File: controllers/builders/generate_config_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			output, err := convertRawExtensionToYaml(tt.input)
+			if tt.wantErr {
+				assert.NotNil(t, err)
+			} else {
+				assert.Nil(t, err)
```
Comment (kannon92) [inline controllers/builders/generate_config_test.go:43]:
```suggestion
				assert.NoError(t, err)
```

**Review by kannon92 — COMMENTED** [review]
I think everything looks good minus the linter errors.  I think we were green on that so can you run `make lint` and fix those before we merge.

File: controllers/builders/generate_config_test.go (hunk 3 above)
Comment (suprjinx) [reply controllers/builders/generate_config_test.go:30]: ghack?

File: apis/install/v1alpha1/common.go (hunk 2 above)
Comment (suprjinx) [reply apis/install/v1alpha1/common.go:23]: removing -- "repository:tag" are the only real naming attributes

**Pull request merged**